movement/movement.c

```diff
@@ -462,6 +468,7 @@ void app_wake_from_standby(void) {
 
 static void _sleep_mode_app_loop(void) {
     movement_state.needs_wake = false;
+    movement_state.ignore_alarm_btn_after_sleep = true;
```

Comment:
Does this need to be configurable? Is there a case where not ignoring erroneous bouncing button presses on watch wake up is advantageous?

I'd also advocate for it being a compile time configuration with preprocessor macros. This would allow the compiler to dead code eliminate all unused branches that depend on this boolean, reducing code size and memory usage.

Reply:
Is the size of a boolean a concern?
My worry is that more preprocessor macros will add more clutter.
I have found this logic needed as I allow toggling 24Hr mode with the ALARM button and waking the watch will always trigger that.
I cannot think of an event where a person would like to wake the watch and trigger the alarm button at the same time.
But I can ask the Discord.

Reply:
> Is the size of a boolean a concern?

Perhaps. Since it is low power embedded software, I assume that memory is always a concern and am always trying to spot opportunities to reduce runtime state. I had to hot patch an out of memory bug in the TOTP face once...

I have analyzed the `movement_state_t` structure with and without the boolean:

```
WITH BOOLEAN

/*     44      |       1 */    uint8_t debounce_ticks_light;
/*     45      |       1 */    uint8_t debounce_ticks_alarm;
/*     46      |       1 */    uint8_t debounce_ticks_mode;
/*     47      |       1 */    _Bool ignore_alarm_btn_after_sleep;

                               /* total size (bytes):   56 */

WITHOUT BOOLEAN

/*     44      |       1 */    uint8_t debounce_ticks_light;
/*     45      |       1 */    uint8_t debounce_ticks_alarm;
/*     46      |       1 */    uint8_t debounce_ticks_mode;
/* XXX  1-byte hole      */

                               /* total size (bytes):   56 */
```

With the structure's current layout, the boolean simply occupies space that used to be padding. If we sort the structure members by size, we can reduce the structure's size to 48 bytes. This boolean just barely makes it into the structure without additional padding being added, it occupies the very last padding byte:

```
WITH BOOLEAN

/*     45      |       1 */    _Bool has_scheduled_background_task;
/*     46      |       1 */    _Bool needs_wake;
/*     47      |       1 */    _Bool ignore_alarm_btn_after_sleep;

                               /* total size (bytes):   48 */
WITHOUT BOOLEAN

/*     45      |       1 */    _Bool has_scheduled_background_task;
/*     46      |       1 */    _Bool needs_wake;
/* XXX  1-byte padding   */

                               /* total size (bytes):   48 */
```

So it's not causing any impact at the moment. The calculus could change if more state is added. For example, adding another boolean or byte sized variable to the structure would cause its size to increase to 52 bytes.

> My worry is that more preprocessor macros will add more clutter.

Do you think the macro adds clutter? I think they are similar enough in complexity to a boolean variable, perhaps even lower complexity since they don't exist at runtime.

I think if anything adds clutter its this configuration itself. Perhaps we could make the case for it being unnecessary and eliminate both the variable and the macro.

> I cannot think of an event where a person would like to wake the watch and trigger the alarm button at the same time.

Me neither! I think the behavior you proposed where the alarm input is ignored when the watch is woken up is the correct thing to do and that it shouldn't even need to be configured at all. Failing that, I think the next best option is making it a compile time toggle via preprocessor macro.